### ISSUE TITLE
Confusing sharding axes ordering with `jax.typeof`

### ISSUE THREAD

**Issue by mrlazy1708** [jax-ml/jax#32163]
### Description

The axes ordering given by `jax.typeof` of a sharded array is confusing:
```python
import jax
import jax.numpy as jnp

from jax.sharding import PartitionSpec as P

def f(a: jax.Array):
  print(jax.typeof(a))


mesh = jax.make_mesh((2, 3, 4), ax:=("x", "y", "z"))
jax.shard_map(f, mesh=mesh, in_specs=P(*ax), out_specs=None)(jnp.zeros((2, 3, 4)))
```

The output is
```
float32[1,1,1]{x,z,y}
```
which is confusing given that the axis names are `x, y, z`.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.7.2
jaxlib: 0.7.2
numpy:  2.3.3
python: 3.11.13 (main, Jun  5 2025, 13:12:00) [GCC 11.2.0]
device info: cpu-24, 24 local devices"
process_count: 1
platform: uname_result(system='Linux', node=*, release='5.15.0-126-generic', version='#136-Ubuntu SMP Wed Nov 6 10:38:22 UTC 2024', machine='x86_64')
JAX_NUM_CPU_DEVICES=24
JAX_PLATFORMS=cpu
```

**Comment by yashk2810** [comment]
Those are not sharding axes. Those are varying manual axes. See https://docs.jax.dev/en/latest/notebooks/shard_map.html#tracking-how-values-vary-over-manual-mesh-axes-and-check-vma-true for more information.

**Comment by yashk2810** [comment]
The vma's are a set and it's true that we can print them in the order of mesh axes. We in fact need to make sure the order is consistent during lowering, so we might as well do it during printing.

Thank you for asking the question!

**Comment by mrlazy1708** [comment]
Thanks for the clarification and quick fix!
